packages/cli/src/lib/platform/authenticatedClient.ts

```diff
@@ -29,18 +29,18 @@ export class AuthenticatedClient {
     }
   }
 
-  async getClient() {
-    const {
-      environment,
-      region,
-      organization,
-      accessToken,
-    } = await this.cfg.get();
+  async getClient(customConfig?: Partial<Configuration>) {
+    const {environment, region, organization, accessToken} =
```

Comment:
You could merge the the 2 config at the beginning of the method, and then just reference the final config instead of multiple boolean check.

```
const configFromDisk = await this.cfg.get();
const resolvedConfig = {...configFromDisk, ...customConfig}

return new PlatformClient({environment: resolvedConfig.environment, [...etc...]})
```